core/drlibc/drlibc_aarch64.asm

```diff
@@ -125,4 +129,32 @@ GLOBAL_LABEL(dynamorio_mach_syscall:)
         END_FUNC(dynamorio_mach_syscall)
 #endif
 
+#ifdef LINUX
+/* thread_id_t dynamorio_clone(uint flags, byte *newsp, void *ptid, void *tls,
+ *                             void *ctid, void (*func)(void))
+ */
+        DECLARE_FUNC(dynamorio_clone)
+GLOBAL_LABEL(dynamorio_clone:)
+        stp      ARG6, x0, [ARG2, #-16]! /* func is now on TOS of newsp */
+        /* All args are already in syscall registers. */
+        mov      SYSNUM_REG, #SYS_clone
+        svc      #0
+        cbnz     x0, dynamorio_clone_parent
+        ldp      x0, x1, [sp], #16
+        blr      x0
+        /* aarch64 doesn't provide unexpected_return (ref i#4304). Instead of
+         * providing our own copy, just inline its definition here.
```

Comment:
We could have aarch64 include it in drlibc but named `drlibc_unexpected_return` and then reference it here and it would reach, right?  That seems nicer than duplicating it in the caller.

Reply:
Code updated to call unexpected_return